src/components/FromBob/bobConversionUtils.ts

```diff
@@ -0,0 +1,242 @@
+    } else if (visible === "false") {
+      outputProps.visible = false;
+    }
+  } catch (e) {
+    log.error(
+      `Could not convert visible property ${inputProps.visible} to a number`
+    );
+  }
+};
+
+export const bobAvoidStyleProp = (
+  inputProps: UnknownPropsObject,
+  outputProps: UnknownPropsObject
+): void => {};
+
+export const bobActionToAction = (
+  bobAction: UnknownPropsObject
+): WidgetActions => {
+  let actionsToProcess: any[] = [];
+  if (Array.isArray(bobAction)) {
+    actionsToProcess = bobAction;
+  } else {
+    actionsToProcess = [bobAction];
+  }
+
+  // Object of available actions
+  const availableActions: { [key: string]: any } = {
+    write_pv: WRITE_PV, // eslint-disable-line @typescript-eslint/camelcase
+    WRITE_PV: WRITE_PV
+  };
+
+  // Turn into an array of Actions
+  let processedActions: WidgetActions = { executeAsOne: false, actions: [] };
+
+  actionsToProcess.forEach((action): void => {
+    log.debug(action);
+    let type: string = availableActions[action._attributes.type];
+    if (type === WRITE_PV) {
+      processedActions.actions.push({
+        type: WRITE_PV,
+        pvName: action.pv_name._text,
+        value: action.value._text,
+        description: action.description._text
+      });
+    }
+  });
+
+  return processedActions;
+};
+
+export const bobHandleActions = (
+  inputProps: UnknownPropsObject,
+  outputProps: UnknownPropsObject
+): void => {
+  if (inputProps.actions.action) {
+    outputProps.actions = bobActionToAction(inputProps.actions.action);
+  }
+};
+
+export const bobChildToWidgetChild = (
+  bobChild: BobDescription,
+  functionSubstitutions?: FunctionSubstitutionInterface,
+  keySubstitutions?: { [key: string]: any }
+): WidgetDescription => {
+  // Convert a non-root widget from the bob file into a widget
+  // It is passed as a JS object now
+  // Extract useful props
+  const {
+    _attributes,
+    x,
+    y,
+    height,
+    width,
+    widget = [],
+    ...remainingProps
+  } = bobChild;
+
+  // Map the remaining props
+  // Checks that there is a substitution map
+  let mappedProps: { [key: string]: any } = {};
+  Object.entries(remainingProps as UnknownPropsObject).forEach(
+    ([key, value]): void => {
+      if (functionSubstitutions && functionSubstitutions[key]) {
+        // Use the function substitution
+        functionSubstitutions[key](remainingProps, mappedProps);
+      } else if (keySubstitutions && keySubstitutions[key]) {
+        // Just substitute the key and extract from _text
+        mappedProps[keySubstitutions[key]] = value._text;
+      } else {
+        // Just extract from text
+        mappedProps[key] = value._text;
+      }
+    }
+  );
+
+  // Make sure widget is an array, otherwise it is seen as an object only
+  let widgetList = [];
+  if (Array.isArray(widget)) {
+    widgetList = widget;
+  } else {
+    widgetList = [widget];
+  }
+
+  // Check that the primary props were defined or use a default value
```

Comment:
Make a note about default values in bob files, this is a technical detail which will need implementation at a later date

Reply:
Done